### ISSUE TITLE
Intel Gfx Geometry shaders no longer work, bisected to 1001-1003

### ISSUE THREAD

**Issue by Beherith** [beyond-all-reason/spring#424]
Geometry shaders, such as healthbars and selection platters which use SUniformBuffers no longer display after commit 1003. It still works in 1001.

Broken by: https://github.com/beyond-all-reason/spring/commit/6c7e8464ac8002e80bc04c3f77ec24a54c5978a8

Would you like me to get an infolog with /debuggl for both?

To repro: launch game, select commander, see no selection platter, start building, see no healthbars.
'''
struct SUniformsBuffer {
    uint composite; //     u8 drawFlag; u8 unused1; u16 id;
    
    uint unused2;
    uint unused3;
    uint unused4;
    float maxHealth;
    float health;
    float unused5;
    float unused6;
    
    vec4 speed;    
    vec4[5] userDefined; //can't use float[20] because float in arrays occupies 4 * float space
};
layout(std140, binding=1) readonly buffer UniformsBuffer {
    SUniformsBuffer uni[];
}; 
'''

**Comment by lhog** [comment]
We are walking on the shaky grounds if the driver doesn't report GL4, yet allows some of its extensions.
Send the infolog, I'm curious to see what they support.

Currently GL4 is determined as:
```
	haveGL4 = static_cast<bool>(GLEW_ARB_multi_draw_indirect);
	haveGL4 &= static_cast<bool>(GLEW_ARB_uniform_buffer_object);
	haveGL4 &= static_cast<bool>(GLEW_ARB_shader_storage_buffer_object);
	haveGL4 &= (globalRenderingInfo.glslVersionNum >= 430); // #version 430 is what engine shaders use
	haveGL4 &= !forceDisableGL4;
```

**Comment by Beherith** [comment]
[infolog.txt](https://github.com/beyond-all-reason/spring/files/9664637/infolog.txt)

**Comment by lhog** [comment]
So it has UBO & SSBO and no GL/GLSL support. Probably ARB_multi_draw_indirect support, who knows

So we're in a peculiar state here:
This machine can't run native GL4 engine shaders so it makes no sense to upload the data, from another point of view if shaders version is downgraded and a bunch of extensions is enabled, perhaps it will run.

I'm gonna expose ARB_multi_draw_indirect detection to infolog and let's see if we can find some middle-ground for Intel

**Comment by lhog** [comment]
https://github.com/beyond-all-reason/spring/commit/35b26b162fccb72ac3418fb899f59bbd9db41dfc

Let's hack engine's shaders when you have some time available

**Comment by Beherith** [comment]
Sure, what do we need to do with the engine shaders?

**Comment by lhog** [comment]
Can you check HEAD if Intel driver can recognize itself as GL4 capable?

**Comment by Beherith** [comment]
I couldnt find the keyword multidraw in this log from 1223

**Comment by lhog** [comment]
> [t=00:00:00.602224] multi draw indirect : 1 (-)
> [t=00:00:00.601473] GL4 support : 1

**Comment by Beherith** [comment]
does it ever, its potatoes are more capable, unlike amd potatoes

**Comment by lhog** [comment]
Is this not fixed btw?

**Comment by Beherith** [comment]
Fixed in 4fae6f51860e831ef27b84800ba2f244e779408b thanks!
